xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.xml.codec;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.codec.CodecConfiguration;
+import io.micronaut.jackson.codec.JacksonMediaTypeCodec;
+import io.micronaut.jackson.codec.JacksonFeatures;
+import io.micronaut.runtime.ApplicationConfiguration;
+
+import javax.annotation.Nullable;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+/**
+ * A jackson based {@link io.micronaut.http.codec.MediaTypeCodec} that handles XML requests/responses.
+ *
+ * @author Sergey Vishnyakov
+ * @since 1.3.0
+ */
+@Named("xml")
+@Singleton
+@BootstrapContextCompatible
+public class XmlMediaTypeCodec extends JacksonMediaTypeCodec {
+
+    public static final String CONFIGURATION_QUALIFIER = "xml";
+
+    /**
+     * @param xmlMapper                Object mapper for xml. If null, retrieved from beanContext
+     * @param applicationConfiguration The common application configurations
+     * @param codecConfiguration       The configuration for the codec
+     */
+    public XmlMediaTypeCodec(@Named(CONFIGURATION_QUALIFIER) ObjectMapper xmlMapper,
```

Comment:
So developers are able to customize ObjectMapper for json, but not for xml? if both media type are processed with Jackson IMHO `@JackongFeatures` should be applicable for both media types

Reply:
Where are you getting the impression that it isn't?

Reply:
I meant customization using `@JacksonFeatures` annotation that is applied on the client and only processed within `HttpClientIntroductionAdvice`. We use annotation data to customize objectMapper for `json` media type,  and not doing same for `xml`

Reply:
ahh see now, false alarm :)